src/Swift.Runtime/src/Metadata/ITypeMetadataCache.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// Represents an interface that defines how a cache for TypeMetadata should operate.
+/// The implementation of this interface needs to be thread safe.
+/// </summary>
+public interface ITypeMetadataCache
+{
+    /// <summary>
+    /// Returns true if and only if the cache contains an entry for t
+    /// </summary>
+    /// <param name="type">The type to look up in the cache</param>
+    /// <returns>true if and only if the cache contains t, false otherwise</returns>
+    bool Contains(Type type);
```

Comment:
What is the intended usage of this? Give that we're introducing a public API here, we should use a pretty high bar on its necessity and quality.

Reply:
I'll think on this. I don't know yet if it will fall into the category of YAGNI (you ain't gonna need it) or not. `GetOrAdd` combines Contains, Get, and Add into one method and `TryGet` combines `Contains` and `Get` into one method, so it might not be needed. I suppose I can take it out and we can always add it later.